src/lib/data/dataview/category/CategorySourceDataView.ts

```
@@ -0,0 +1,112 @@
+
+    const { operation, operationColumn, limit } = options || {};
+
+    validateParameters(operation, operationColumn);
+
+    this._operation = operation;
+    this._operationColumn = operationColumn;
+    this._limit = limit;
+
+    this.on('optionChanged', () => {
+      // timeout prevents multiple calls to getData
+      // when user sets several options in a row
+      if (this.optionChangedTimeoutId) {
+        window.clearTimeout(this.optionChangedTimeoutId);
+      }
+
+      this.optionChangedTimeoutId = window.setTimeout(async () => {
+        const newData = await this.getData();
+        this.emit('dataChanged', [newData]);
+      }, OPTION_CHANGED_DELAY);
+    });
+  }
+
+  public set operation(operation: AggregationType) {
+    this._operation = operation;
+    this.emit('optionChanged');
+  }
+
+  public set operationColumn(operationColumn: string) {
+    this._operationColumn = operationColumn;
+    this.emit('optionChanged');
+  }
+
+  public set limit(limit: number) {
+    this._limit = limit;
+    this.emit('optionChanged');
+  }
+
+  public async getData() {
+    const aggregationResponse = await this.dataviewsApi.aggregation({
+      column: this.column,
+      aggregation: this._operation,
+      operationColumn: this._operationColumn,
+      limit: this._limit
+    });
+
+    if (
+      aggregationResponse.errors_with_context &&
+      aggregationResponse.errors_with_context.length > 0
+    ) {
+      this.emit('error', aggregationResponse.errors_with_context);
+      const { message, type } = aggregationResponse.errors_with_context[0];
+      throw new CartoDataViewError(`${type}: ${message}`, dataViewErrorTypes.MAPS_API);
+    }
+
+    const { aggregation, categories, count, max, min, nulls } = aggregationResponse;
+
+    const adaptedCategories = categories.map(({ category, value }) => {
+      return {
+        name: category,
+        value
+      };
+    });
```

I think we should just throw an error (and catch it in the client), instead of using an evented error approach... for these dataviews and in general for the whole lib. Aggree on this @jesusbotella ?